### DISCUSSION TITLE
Adding gene-partnership appyter URL on the CFDE portal gene pages

### DISCUSSION THREAD

**Discussion by mano-at-sdsc** [nih-cfde/published-documentation#381] [Feature Requests] [answered]
Dear CC team:

We think it will be useful to have a link to the gene-partnership appyter 
https://appyters.maayanlab.cloud/CFDE-Gene-Partnership/
[ending / must be included in the URL] on the gene pages on the CFDE portal.  The appyter URL can be listed above the Phenotype table and below the Description line. Thank you.

Mano

**Comment by ctb** [comment]
hi @mano-at-sdsc just to check - for now, you're asking for this general link to be added to each gene page, without specifically linking to the gene that page describes?

**Comment by AviMaayan** [comment]
Hi @ctb there is a way to make it a button/link that will execute the Appyter for each gene.

**Comment by ctb** [comment]
specific information on how to do this properly would be a welcome addition to this discussion :)

backing up a bit,

right now we're building out workflows to update the content registry, and we cannot quickly or robustly add customized information per gene. that will be straightforward within a month or so, we expect.

we might be able to include a link to this appyter sooner, but not in a customized way - it would just be a straight link to that appyter as I interpret was requested above. That seems to me to be less useful and quite possibly misleading.

I'll assume you want the custom embedding unless you say otherwise.

**Comment by AviMaayan** [comment]
@mano-at-sdsc @ctb I think only the customized embedding would be useful. The Appyter itself needs to be further polished. We'll provide instructions on how to make the customized embedding.

**Comment by mano-at-sdsc** [comment]
@AviMaayan & @ctb If the gene-specific URL can be generated, for example, 
https://appyters.maayanlab.cloud/CFDE-Gene-Partnership/c290434dfa28134d695be123c095e146cf6e5d26/, for ACE2, then that will be the best. If not for now, then at least we can embed "https://appyters.maayanlab.cloud/CFDE-Gene-Partnership/" so that upon clicking the user can type the gene symbol (or other IDs accepted by the appyter) and execute the appyter upon "Submit".

@AviMaayan I suppose you meant you will provide the steps to generate the code, e.g., "(https://appyters.maayanlab.cloud/CFDE-Gene-Partnership/c290434dfa28134d695be123c095e146cf6e5d26/)", for ACE2; "https://appyters.maayanlab.cloud/CFDE-Gene-Partnership/1d6e8088bb28a63bccb78ef068654b0afab324f5/", for PNPLA3, etc.

**Reply by ctb** [reply]
yes, we would need scripting code (python pref, or command line if not) for going from gene name to the hash.

also, is there a list of canonical gene names that are acceptable? or can we simply use whatever gene names are in the C2M2 controlled vocabulary list?

**Reply by u8sand** [reply]
@ctb The C2M2 controlled vocabulary should be fine, ensembl ids are supported as are a number of others. And you don't build the hash, the app does, you can use:
https://appyters.maayanlab.cloud/CFDE-Gene-Partnership/#?args.gene=ACE2&submit

Replacing ACE2 with the ENSEMBL ID; the front-end will translate this into the necessary POST when someone visits the link.

NOTE: We just added this feature, so if it's not working for you, be sure to clear your browser cache.

**Reply by AviMaayan** [reply]
Courtesy of @u8sand you can have a link formatted this way: https://appyters.maayanlab.cloud/CFDE-Gene-Partnership/#?args.gene=STAT3&submit to invoke the execution of this Appyter with the specific gene name.

**Reply by ctb** [reply]
(that doesn't seem to work for me, @AviMaayan)

**Reply by AviMaayan** [reply]
Just need to clear the browser cache... SHIFT+refresh... It happened to me too...

**Reply by ctb** [reply]
excellent, thanks! works for me in a private browser window.

**Comment by mano-at-sdsc** [comment]
I see the appyter can take Official gene symbol, available in the C2M2 controlled vocabulary list. @AviMaayan can provide further details about if the appyter prefers one type of ID vs. another. Gene synonyms should be available on the portal, e.g., https://app.nih-cfde.org/chaise/record/#1/CFDE:gene/nid=1, at least for some genes [I had shared a table of different ID types (Arthur and Amanda were on that email I think)].

**Comment by ctb** [comment]
Hi folks,

thanks to a lot of work by @abradyIGS and others, we've got a general workflow going for updating the content registry in a robust way, and I used the gene-partnership appyter URL as an example.

Here's a screenshot from the dev site - seem ok? Updated text welcome. Style suggestions may ned wait, I'm still getting the hang of the specific Markdown syntax :)

<img width="1187" alt="Screen Shot 2022-06-27 at 4 33 40 AM" src="https://user-images.githubusercontent.com/51016/175849644-3ddab8f6-8d3a-43c6-ac83-550e8d2e6f37.png">


Also, right now there's no simple way to give you access to it, or to generate previews, but we're working on that!

**Comment by ctb** [comment]
the link leads to https://appyters.maayanlab.cloud/CFDE-Gene-Partnership/#?args.gene=ENSG00000000457&submit

**Reply by AviMaayan** [reply]
It would be great if you can change "appyter" to "Appyter"; and the text below to: "This Appyter provides up-to-date information on human genes from across the Common Fund Data Ecosystem. The Appyter collects gene-centric data from Common Fund supported programs via their API. The source code within the Appyter demonstrates how you can programmatically access CFDE data sources for integrative analyses."

**Comment by ctb** [comment, marked as answer]
done!

<img width="1173" alt="Screen Shot 2022-06-27 at 7 18 28 PM" src="https://user-images.githubusercontent.com/51016/175998673-cba811f0-7522-4402-8b28-95f1c9b9b739.png">

we don't yet have a schedule for when this will be live on the portal site, but we will have the schedule soon after the end of this week, and it won't be too far out.

**Reply by AviMaayan** [reply]
Looks good. Thank you.
